### DISCUSSION TITLE
Avalonia exception handler?

### DISCUSSION THREAD

**Discussion by josephnarai** [AvaloniaUI/Avalonia#7988] [General]
Hi All,
   
   I've created a handler for Unhandled exceptions which saves the exception info to a file on the users desktop that they can then send me if my Avalonia app crashes.
   
   I have hooked into AppDomain.CurrentDomain.UnhandledException and TaskScheduler.UnobservedTaskException.

   The first seems to have caught the typical crashes I've been able to test for.

   I note that there are some other exceptions you can hook into in other frameworks for example Applicaiton.ThreadException.

   Is there any Avalonia specific exception handling I should hook into?

   Thanks,

J

**Comment by maxkatz6** [comment]
Avalonia UI thread is the main thread of the application. So, you can catch them with "try catch" in the Main method of your program.
TaskScheduler.UnobservedTaskException is useful for unhandled exceptions in any threadpooled task.
And if you use ReactiveUI - ```RxApp.DefaultExceptionHandler```

See https://github.com/AvaloniaUI/Avalonia/issues/5290

**Reply by josephnarai** [reply]
Thanks, as I mentioned I've implemented AppDomain.CurrentDomain.UnhandledException and TaskScheduler.UnobservedTaskException. and it seems to working perfectly. Thank you.

**Reply by timunie** [reply]
@maxkatz6 do you think we should add this into the docs? Seems to be frequently asked. If so, I can pick it up when I find a free minute

**Reply by maxkatz6** [reply]
@timunie yeah, it makes sense.

**Reply by dotNET-anykey** [reply]
> @maxkatz6 do you think we should add this into the docs? Seems to be frequently asked. If so, I can pick it up when I find a free minute

Any success on writing documentation how to use TaskScheduler.UnobservedTaskException?

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/getting-started/unhandledexceptions anything missing or unclear in this particular page?

**Reply by sn4k3** [reply]
>  anything missing

In addition to Main thread and `TaskScheduler.UnobservedTaskException` I also use `AppDomain.CurrentDomain.UnhandledException`

**Reply by timunie** [reply]
added to our v11 documentation todo-list. So should be added to docs soon.

**Reply by tfojt** [reply]
Hey guys, I just wanted to check the documentation, if anything new was added by any chance. Has the documentation been moved somewhere else, or removed? Thanks!

![image](https://github.com/AvaloniaUI/Avalonia/assets/46399943/7308b1f8-dd4b-4476-ba28-52607b5859a3)

**Reply by maxkatz6** [reply]
@tfojt see https://docs.avaloniaui.net/docs/getting-started/unhandled-exceptions
Thanks for letting us know, some other links needs to be double checked.
